### PR TITLE
Fix the target branch setting and validation

### PR DESCRIPTION
**Pull request by sima-zhu**
https://dev.azure.com/azure-sdk/public/_build/results?buildId=1458044&view=logs&j=b70e5e73-bbb6-5567-0939-8415943fadb9&t=c9190431-7417-53a5-3fc1-7c7e026cd79d

Testing pipeline on eng/common changes in lang-repo: https://dev.azure.com/azure-sdk/public/_build/results?buildId=1458809&view=logs&s=6884a131-87da-5381-61f3-d7acc3b91d76

image

Testing on no eng/common changes in lang-repo:
https://dev.azure.com/azure-sdk/public/_build/results?buildId=1458993&view=logs&j=76992fd1-2312-5fae-7c45-7baba86b87ae
Testing on manually triggers:
https://dev.azure.com/azure-sdk/public/_build/results?buildId=1458994&view=results

### CODE REVIEW

File: eng/common/pipelines/templates/steps/eng-common-workflow-enforcer.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@ steps:
     - pwsh: |
         # Find the default branch of the repo. The variable value sets in build step.
         Write-Host "Default Branch: $(DefaultBranch)"
-        $targetBranch = "$(System.PullRequest.TargetBranch)" -replace "refs/heads/"
+        $targetBranch = "origin/$(System.PullRequest.TargetBranch)" -replace "refs/heads/"
 
         if ((!"$(System.PullRequest.SourceBranch)".StartsWith("sync-eng/common")) -and $targetBranch -eq "$(DefaultBranch)")
```
Comment (weshaggard) [inline eng/common/pipelines/templates/steps/eng-common-workflow-enforcer.yml:13]:
```suggestion
        if ((!"$(System.PullRequest.SourceBranch)".StartsWith("sync-eng/common")) -and "$(System.PullRequest.TargetBranch)"-eq "$(DefaultBranch)")
```

File: eng/common/pipelines/templates/steps/eng-common-workflow-enforcer.yml [hunk 2]
```diff
@@ -8,9 +8,9 @@ steps:
     - pwsh: |
         # Find the default branch of the repo. The variable value sets in build step.
         Write-Host "Default Branch: $(DefaultBranch)"
-        $targetBranch = "$(System.PullRequest.TargetBranch)" -replace "refs/heads/"
+        $targetBranch = "origin/$(System.PullRequest.TargetBranch)" -replace "refs/heads/"
 
-        if ((!"$(System.PullRequest.SourceBranch)".StartsWith("sync-eng/common")) -and $targetBranch -eq "$(DefaultBranch)")
+        if ((!"$(System.PullRequest.SourceBranch)".StartsWith("sync-eng/common")) -and "$(System.PullRequest.TargetBranch)" -match "(refs/heads)?$(DefaultBranch)"))
```
Comment (weshaggard) [inline eng/common/pipelines/templates/steps/eng-common-workflow-enforcer.yml:13]:
```suggestion
        if ((!"$(System.PullRequest.SourceBranch)".StartsWith("sync-eng/common")) -and "$(System.PullRequest.TargetBranch)" -match "^(refs/heads)?$(DefaultBranch)$"))
```

File: eng/common/pipelines/templates/steps/eng-common-workflow-enforcer.yml [hunk 3]
```diff
@@ -8,11 +8,10 @@ steps:
     - pwsh: |
         # Find the default branch of the repo. The variable value sets in build step.
         Write-Host "Default Branch: $(DefaultBranch)"
-        $targetBranch = "$(System.PullRequest.TargetBranch)" -replace "refs/heads/"
 
-        if ((!"$(System.PullRequest.SourceBranch)".StartsWith("sync-eng/common")) -and $targetBranch -eq "$(DefaultBranch)")
+        if ((!"$(System.PullRequest.SourceBranch)".StartsWith("sync-eng/common")) -and "$(System.PullRequest.TargetBranch)" -match "^(refs/heads)?$(DefaultBranch)$")
```
Comment (weshaggard) [inline eng/common/pipelines/templates/steps/eng-common-workflow-enforcer.yml:12]:
```suggestion
        if ((!"$(System.PullRequest.SourceBranch)".StartsWith("sync-eng/common")) -and "$(System.PullRequest.TargetBranch)" -match "^(refs/heads/)?$(DefaultBranch)$")
```

Comment (sima-zhu) [reply eng/common/pipelines/templates/steps/eng-common-workflow-enforcer.yml]: Done

**Review by weshaggard — APPROVED** [review]

**Pull request merged**